qiskit/transpiler/__init__.py

```diff
@@ -40,714 +41,1131 @@
 Qiskit has four pre-built transpilation pipelines available here:
 :mod:`qiskit.transpiler.preset_passmanagers`.  Unless the reader is familiar with
 quantum circuit optimization methods and their usage, it is best to use one of
-these ready-made routines.
+these ready-made routines. By default the preset pass managers are composed
+of 6 stages:
+
+#. ``init`` - This stage runs any initial passes that are run before we start embedding the
+   circuit to the backend
```

Comment:
Is there an example we can add here of how this stage is used in the preset pass managers, since there are no further details (e.g. checking for transpiler support of instruction types in the circuit)?

Reply:
For the preset pass managers the only thing the init stage does right now is check if there are control flow instructions in the circuit and in the case and unroll >2q instructions to 1 and 2q instructions so that layout and routing can reason about them (which will potentially include the unitary synthesis and HighLevelSynthesis passes).

Reply:
Can we add something to that effect? "For example, in the preset pass managers, this stage is used to check compatibility of input instructions with the different passes in the transpiler, and convert custom instructions to a standard form." This reminds me we should probably carve out some of these passes into a canonicalization stage.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9087/commits/b516ca9efb36038879810e80b3974fa47100e9d4